Kernel/Net/NetworkTask.cpp

```diff
@@ -202,7 +202,7 @@ void handle_ipv4(EthernetFrameHeader const& eth, size_t frame_size, Time const&
     dbgln_if(IPV4_DEBUG, "handle_ipv4: source={}, destination={}", packet.source(), packet.destination());
 
     NetworkingManagement::the().for_each([&](auto& adapter) {
-        if (adapter.link_up()) {
+        if (adapter.link_up() && (adapter.ipv4_address().to_u32() != 0)) {
```

Comment:
nitpick, but can we flip the condition to reduce the level of indentation a bit?

Reply:
Sure thing. Actually, we can leverage AK/IPv4Address here too. How about this?

`if (adapter.link_up() && !adapter.ipv4_address().is_zero())`

Reply:
Sure, with the condition flipped though :^)

Reply:
Haha I'm having a slow moment. Are you saying like this?

`if (!adapter.ipv4_address().is_zero() && adapter.link_up())`

Or do you mean just put it in a separate conditional statement in the beginning and just return immediately:

```
 if (adapter.ipv4_address().is_zero())
    return
```

Reply:
The second option, but include `|| !adapter.link_up()` as well, the `return early` pattern usually results in flatter and easier-to-understand code IMO

Reply:
Let me know if that's better. Thanks!